numba/cuda/cudadrv/driver.py

```diff
@@ -203,7 +203,9 @@ def __init__(self):
         self.pid = None
         try:
             if config.DISABLE_CUDA:
-                raise CudaSupportError("CUDA disabled by user")
+                msg = ("CUDA disabled by user. Disabled due to NUMBA_DISABLE_CUDA=1 or "
+                       "on a 32-bit system, which CUDA doesn't support")
```

Comment:
I think we can streamline this message to "CUDA is disabled due to setting NUMBA_DISABLE_CUDA=1 in the environment, or because CUDA is unsupported on 32-bit systems."